### DISCUSSION TITLE
Examples of Generative Model

### DISCUSSION THREAD

**Discussion by nashid** [pyg-team/pytorch_geometric#2614] [Ideas]
I am not sure whether this is the correct channel to initiate discussion to should I use the GitHub issues!

I find Pytorch Geometric is easy to use and there are great examples for graph classification, matching i.e. predictive tasks. However, Pytorch Geometric lacks examples of generative models. 

I am wondering whether there are any plans to include any of the following models:

- GraphRNN: https://github.com/JiaxuanYou/graph-generation

Furthermore, Pytorch Geometric lacks examples of Graph2Seq based models. It would be awesome to include consider any of the following Graph2Seq based generative models.

- Graph-to-Sequence Learning using Gated Graph Neural Networks, ACL’18
https://github.com/beckdaniel/acl2018_graph2seq
Using MXNet

- Densely Connected Graph Convolutional Networks for Graph-to-Sequence Learning, ACL’19
https://github.com/Cartus/DCGCN
Using MXNet 1.3.0

- Heterogeneous Graph Transformer for Graph-to-Sequence Learning, ACL’18 
https://github.com/QAQ-v/HetGT
Using OpenNMT

**Comment by rusty1s** [comment]
Yes, this is always something that I wanted to add, but never had the time to do so. I would very much appreciate an initiative to start adding generative models to the library.

**Reply by orausch** [reply]
I opened a PR here: https://github.com/pyg-team/pytorch_geometric/pull/4398
